graspologic/plot/plot.py

```diff
@@ -671,6 +673,229 @@ def pairplot(
+    if labels is not None:
+        check_X_y(X, labels)
+    # Handle gmm
+    if gmm is None:
+        msg = "You must input a sklearn.mixture.GaussianMixture"
+        raise NameError(msg)
+    Y_, means, covariances = gmm.predict(X), gmm.means_, gmm.covariances_
+    data = pd.DataFrame(data=X)
+    n_components = gmm.n_components
+    # setting up the data DataFrame
+    if labels is None:
+        lab_names = [i for i in range(n_components)]
+        data["labels"] = np.asarray([lab_names[Y_[i]] for i in range(Y_.shape[0])])
+    else:
+        data["labels"] = labels
+    data["clusters"] = Y_
+    # labels are given we must check whether input is correct
+    if labels is not None:
+        if isinstance(label_palette, str):
+            colors = sns.color_palette(label_palette, n_components)
+            label_palette = dict(zip(np.unique(np.asarray(labels)), colors))
+        elif not isinstance(label_palette, dict):
+            msg = "When giving labels must supply palette in string or dictionary"
+            raise ValueError(msg)
+        if isinstance(cluster_palette, str):
+            colors = sns.color_palette(cluster_palette, n_components)
+            cluster_palette = dict(zip(np.unique(Y_), colors))
+        elif not isinstance(label_palette, dict):
+            msg = "When giving labels must supply palette in string or dictionary"
+            raise ValueError(msg)
+    else:
+        # no labels given we go to default.
+        colors = sns.color_palette(cluster_palette, n_components)
+        labels = np.unique(Y_)
+        cluster_palette = dict(zip(np.unique(labels), colors))
+        label_palette = dict(zip(np.unique(np.asarray(labels)), colors))
+
+    with sns.plotting_context(context=context, font_scale=font_scale):
+        dimensions = X.shape[1]
+        # we only want 1 scatter plot for 2 features
+        if X.shape[1] == 2:
+            dimensions = 1
+        pairplot, axes = plt.subplots(
+            dimensions, dimensions, figsize=figsize, squeeze=False
+        )
+        # this will allow for uniform iteration whether axes was 2d or 1d
+        axes = axes.flatten()
+        for i in range(dimensions):
+            for j in range(dimensions):
+                if i == j and X.shape[1] > 2:
+                    # take care of the distplot on diagonal
+                    for t, lab in zip([i for i in range(X.shape[1])], label_palette):
+                        sns.distplot(
+                            X[Y_ == t, i],
+                            kde=kde,
+                            ax=axes[dimensions * i + j],
+                            color=label_palette[lab],
+                        )
+                    # this removes the tick marks from the histplot
+                    axes[dimensions * i + j].set_xticks([])
+                    axes[dimensions * i + j].set_yticks([])
+                else:
+                    # take care off off-diagonal scatterplots
+                    ax1, ax2 = j, i
```

Comment:
ax as a variable name is confusing to me here

Reply:
@bdpedigo what would be more appropriate?

Reply:
what was wrong with i and j?

Reply:
@bdpedigo it was to make the code more concise because you see later on i force it to be 0 and 1 if we are with only 2 dims.  but i feed into the plot_ellipse function ax1, ax2

Reply:
dim1, dim2?

Reply:
FYI this is purely cause ax is always a matplotlib Axis in most code so it tripped me up when i was reading it.

Reply:
oh ok! will change to dim1, dim2!

Reply:
https://github.com/microsoft/graspologic/pull/541/commits/2e5c9cd8fb6d759b25aa04392bfc87abc4fffb7c